IPython/terminal/shortcuts.py

```diff
@@ -267,15 +267,22 @@ def ebivim():
     focused_insert_vi = has_focus(DEFAULT_BUFFER) & vi_insert_mode
 
     # Needed for to accept autosuggestions in vi insert mode
-    @kb.add("c-e", filter=focused_insert_vi & ebivim)
-    def _(event):
+    def _apply_autosuggest(event):
         b = event.current_buffer
         suggestion = b.suggestion
         if suggestion:
             b.insert_text(suggestion.text)
         else:
             nc.end_of_line(event)
 
+    @kb.add("end", filter=has_focus(DEFAULT_BUFFER) & ebivim)
```

Comment:
The `ebvim` here states, that if you disable `TerminalInteractiveShell.emacs_bindings_in_vi_insert_mode`, the `End` key stops working in non-vi input mode too, which I think is something we don't want.

The filter here should check if it's in non-vi input mode - if yes, then apply the binding unconditionally.
If we're using Vi bindings, apply it only when `ebvim` is `True`.

Reply:
See [prompt toolkit's builtin filters](https://python-prompt-toolkit.readthedocs.io/en/master/pages/advanced_topics/filters.html#built-in-filters) - I think you could use `~vi_mode` to get the right condition.

Reply:
So this ended as `(ebivim | ~vi_insert_mode)`. It feels kind of wrong to me because in vi command mode it still evaluates to true (AFAIK). I wonder if it would be better as `(vi_insert_mode & ebivim) | emacs_insert_mode`. @MrMino do you happen to still remember if it was considered/would have any disadvantage?